src/background.js

```diff
@@ -1,5 +1,209 @@
+    console.info("Writing hard-coded provider cfr pref", { bucket, cohort });
+    await browser.privileged.messagingSystem.setASRouterCfrProviderPref(
+      bucket,
+      cohort,
+    );
+    return;
+  }
+
+  console.info("Writing hard-coded score threshold", config.scoreThreshold);
+  await browser.privileged.personalizedCfrPrefs.setScoreThreshold(
+    config.scoreThreshold,
+  );
+};
+
+const everyRun = async () => {
+  try {
+    // Do nothing on every run in control branch
+    if (config.branch === "control") {
+      return;
+    }
+
+    // Imports models from remote settings (`cfr-models` bucket)
+    // Drives update cycle so that the computations gets updated on each update
+    console.info("Subscribing to model updates from remote settings");
+    browser.privileged.messagingSystem.onCfrModelsSync.addListener(
+      onCfrModelsSync,
+    );
+
+    // Force sync after 10 seconds and then every 60 minutes
+    const alarmListener = async alarm => {
+      if (alarm.name === periodicAlarmName) {
+        console.info(`Force syncing "cfr-models" bucket contents`);
+        await browser.privileged.remoteSettings.clearLocalDataAndForceSync(
+          "cfr-models",
+        );
+      }
+    };
+    browser.alarms.onAlarm.addListener(alarmListener);
+    const delayInMinutes = (1 / 60.0) * 10; // 10 seconds
+    const periodInMinutesOverride = await browser.privileged.testingOverrides.getPeriodicPollingPeriodInMinutesOverride();
+    const periodInMinutes = periodInMinutesOverride || 60.0;
+    browser.alarms.create(periodicAlarmName, {
+      delayInMinutes,
+      periodInMinutes,
+    });
+  } catch (e) {
+    onError(e);
+  }
+};
+
+const onUnenroll = async reason => {
+  console.info("Unenrolling", { reason });
+  try {
+    await browser.privileged.messagingSystem.clearASRouterCfrProviderPref();
+
+    // Do nothing more in control branch
+    if (config.branch === "control") {
+      return;
+    }
+
+    await browser.alarms.clear(periodicAlarmName);
+    browser.privileged.messagingSystem.onCfrModelsSync.removeListener(
+      onCfrModelsSync,
+    );
+    await browser.privileged.personalizedCfrPrefs.clearScoreThreshold();
+    await browser.privileged.personalizedCfrPrefs.clearScores();
+  } catch (e) {
+    console.error("Error occurred in the unenroll callback");
+    console.error(e);
+  }
+};
+
+const onCfrModelsSync = async syncEvent => {
+  try {
+    console.log("onCfrModelsSync", { syncEvent });
+    const { data } = syncEvent;
+    console.log({ data });
+
+    console.info("Getting current messages from `cfr-experiment`");
+    const cfrExperimentMessages = await browser.privileged.messagingSystem.getCfrBucketMessages(
+      bucket,
+      cohort,
+    );
+    console.log({ cfrExperimentMessages });
+
+    console.info("Getting model inputs from ASRouterTargeting.jsm");
+    const asRouterTargetingGetters = await browser.privileged.messagingSystem.getASRouterTargetingGetters(
```

Comment:
(I'm assuming this is a placeholder until we determine what we actually need)

Reply:
We might want to consider using `requestIdleCallback` if we suspect a big performance overhead here (once we determine what's actually going to be accessed)

Reply:
> We might want to consider using requestIdleCallback if we suspect a big performance overhead here (once we determine what's actually going to be accessed)

Definitely!